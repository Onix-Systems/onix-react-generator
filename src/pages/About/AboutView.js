import React from 'react';
import PropTypes from 'prop-types';
import styles from './sass/About.module.scss';

export default function AboutView({ title }) {
  return (
    <div>
      <p className={styles.appIntro}>{title}</p>
    </div>
  );
}

AboutView.propTypes = {
  title: PropTypes.string.isRequired
};
