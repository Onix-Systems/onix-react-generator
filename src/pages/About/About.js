import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withTranslation } from 'react-i18next';
import AboutView from './AboutView';

class About extends Component {
  render() {
    const { t } = this.props;
    return (
      <AboutView
        title={t('aboutPage')}
      />
    );
  }
}

About.propTypes = {
  t: PropTypes.func.isRequired
};

export default withTranslation()(About);
