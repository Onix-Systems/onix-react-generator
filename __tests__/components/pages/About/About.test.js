import React from 'react';
import { render } from '../../../settings/test-utils';
import About from '../../../../src/pages/About/About';

describe('About />', () => {
  test('should render About component', () => {
    const { asFragment } = render(<About />);
    expect(asFragment(<About />)).toMatchSnapshot();
  });
});
