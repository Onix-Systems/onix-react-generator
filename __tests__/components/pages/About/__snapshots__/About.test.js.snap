// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About /> should render About component 1`] = `
<DocumentFragment>
  <div>
    <p
      class="appIntro"
    >
      About page
    </p>
  </div>
</DocumentFragment>
`;
